stablehlo/dialect/StablehloOps.cpp

```diff
@@ -477,6 +477,24 @@ LogicalResult CustomCallOp::verify() {
              << "operand part has type " << operandPart
              << " and output part has type " << outputPart;
   }
+
+  // Check backend_config attribute.
+  if (auto backendConfig = getBackendConfig()) {
+    if (getApiVersion() == CustomCallApiVersion::API_VERSION_TYPED_FFI) {
+      // Typed FFI custom calls require `backend_config` to be a DictionaryAttr.
+      if (isa<mlir::StringAttr>(*backendConfig))
+        return emitOpError()
+               << "unsupported user-encoded backend config,"
+                  " backend config must be a dictionary attribute.";
```

Comment:
"backend_config for api_version " << api_version << " must be a dictionary"

and below

"backend_config for api_version " << api_version << " must be a string".